core/iwasm/libraries/lib-wasi-threads/tid_allocator.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include "tid_allocator.h"
+#include "wasm_export.h"
+#include "bh_log.h"
+
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+
+struct tidAllocator {
+    int32 *ids;
+    uint32 size;    // Stack capacity
+    uint32 pos;     // Index of the position after the stack top
+    uint32 next_id; // To keep track of the ids to put in the stack
```

Comment:
next_id seems redundant but i guess it isn't critical.

Reply:
I was using it for readability, but probably it's better to get rid of it, being the next id calculation not that complicated